azure/functions/decorators/core.py

```diff
@@ -161,3 +161,40 @@ def __init__(self, name: str, data_type: Optional[DataType] = None,
                  type: Optional[str] = None) -> None:
         super().__init__(direction=BindingDirection.OUT,
                          name=name, data_type=data_type, type=type)
+
+
+class Setting(ABC, metaclass=ABCBuildDictMeta):
+    """ Abstract class for all settings of a function app.
+        This class represents all the decorators that cannot be
+        classified as bindings or triggers. e.g function_name, retry etc.
+    """
+
+    EXCLUDED_INIT_PARAMS = {'self', 'kwargs', 'setting_name'}
+
+    def get_setting_name(self) -> str:
+        return self.setting_name
+
+    def __init__(self, setting_name: str) -> None:
+        self.setting_name = setting_name
+        self._dict: Dict = {
+            "setting_name": self.setting_name
+        }
+
+    def get_dict_repr(self) -> Dict:
+        """Build a dictionary of a particular binding. The keys are camel
+        cased binding field names defined in `init_params` list and
+        :class:`Binding` class. \n
+        This method is invoked in function :meth:`get_raw_bindings` of class
+        :class:`Function` to generate json dict for each binding.
+
+        :return: Dictionary representation of the binding.
+        """
+        params = list(dict.fromkeys(getattr(self, 'init_params', [])))
+        for p in params:
+            if p not in Setting.EXCLUDED_INIT_PARAMS:
+                self._dict[to_camel_case(p)] = getattr(self, p, None)
+
+        return self._dict
+
+    def get_settings_value(self, name: str) -> Optional[str]:
```

Comment:
Change this to something like
```suggestion
    def get_settings_value(self, setting_attribute_key: str) -> Optional[str]:
```

Also add pydoc to explain what this is